backend/grants/admin.py

```diff
@@ -548,88 +552,137 @@ def save_form(self, request, form, change):
+        human_readable_filters = self._format_filters_for_display(filter_params)
+        # Sort the summary data
+        sorted_summary = dict(sorted(summary.items(), key=lambda x: (x[0][0], x[0][2])))
+
+        context = {
+            "summary": sorted_summary,
+            "statuses": statuses,
+            "genders": {code: name for code, name in GENDERS},
+            "total": filtered_grants.count(),
+            "totals_by_status": totals_by_status,
+            "totals_per_continent": totals_per_continent,
+            "gender_summary": gender_summary,
+            "filters": human_readable_filters,
+            **self.admin_site.each_context(request),
+        }
+        return TemplateResponse(request, "admin/grants/grant_summary.html", context)
+
+    def _aggregate_data_by_country(self, grants_data, statuses):
+        """
+        Aggregates grant data by country and status.
+        """
+
+        summary = {}
+        totals_by_status = {status[0]: 0 for status in statuses}
+        totals_per_continent = {}
+
+        for data in grants_data:
+            country = countries.get(code=data["travelling_from"])
+            continent = country.continent.name if country else "Unknown"
+            country_name = f"{country.name} {country.emoji}" if country else "Unknown"
+            country_code = country.code if country else "Unknown"
+            key = (continent, country_name, country_code)
+
+            # Initialize country summary
+            if key not in summary:
+                summary[key] = {status[0]: 0 for status in statuses}
+
+            summary[key][data["status"]] += data["total"]
+            totals_by_status[data["status"]] += data["total"]
+
+            # Update continent totals
+            if continent not in totals_per_continent:
+                totals_per_continent[continent] = {status[0]: 0 for status in statuses}
+            totals_per_continent[continent][data["status"]] += data["total"]
+
+        return summary, totals_by_status, totals_per_continent
+
+    def _aggregate_data_by_gender(self, filtered_grants, statuses):
+        """
+        Aggregates grant data by gender and status.
+        """
+        gender_data = filtered_grants.values("user__gender", "status").annotate(
+            total=Count("id")
+        )
+        gender_summary = {
+            gender: {status[0]: 0 for status in statuses} for gender, _ in GENDERS
+        }
+        gender_summary[""] = {
+            status[0]: 0 for status in statuses
+        }  # For unspecified genders
+
+        for data in gender_data:
+            gender = data["user__gender"] if data["user__gender"] else ""
+            status = data["status"]
+            total = data["total"]
+            gender_summary[gender][status] += total
+
+        return gender_summary
+
+    def _format_filters_for_display(self, filter_params):
+        """
+        Formats raw filter keys to user-friendly names.
```

Comment:
this comment doesn't look useful  as it just explains what's the code after it? 🤔

Reply:
I changed this method and updated the description, let me know if now is more useful :)